src/webln/NostrWeblnProvider.ts

```diff
@@ -291,10 +296,29 @@ export class NostrWebLNProvider implements WebLNProvider, Nip07Provider {
     );
   }
 
-  // not-yet implemented WebLN interface methods
-  keysend(args: KeysendArgs): Promise<SendPaymentResponse> {
-    throw new Error("Method not implemented.");
+  keysend(args: KeysendArgs) {
+    this.checkConnected();
+
+    return this.executeNip47Request<SendPaymentResponse, Nip47PayResponse>(
+      "pay_keysend",
+      {
+        amount: +args.amount * 1000, // NIP-47 uses msat
+        pubkey: args.destination,
+        tlv_records: args.customRecords
+          ? Object.entries(args.customRecords).map((v) => ({
```

Comment:
we need to do this because in the spec the TLV records are formatted as `{type: number, value: string}`